### PR TITLE
chore(splunk_hec sink): Update splunk_hec sink instrumentation

### LINKED ISSUE #10349: Update instrumentation in the `splunk_hec` sinks
Follow up task re: feedback on #10135 to remove/update `splunk_hec` sink internal events according to the component spec.

### PR DESCRIPTION
**Pull request by 001wwang**
Closes #10349 

Todo:
- [x] Update 0.19.0 upgrade guide with details

Following up on instrumentation-related feedback in #10135, this PR:
- Removes the `SplunkEventSent` event. This event was emitted in the wrong place, and `EventsSent` is already emitted by the driver.
- Removes `encode_errors_total` metric to avoid redundancy.
- Updates `SplunkInvalidMetricReceived` to adhere to component error spec.
- Updates the sink `Driver` to emit `EventsSent` only on successful event status

### CODE REVIEW

File: website/cue/reference/components/sources/internal_metrics.cue [hunk 1]
```diff
@@ -509,6 +509,12 @@ components: sources: internal_metrics: {
 			default_namespace: "vector"
 			tags:              _component_tags
 		}
+		component_discarded_events_total: {
```
Comment (jszwedko) [inline website/cue/reference/components/sources/internal_metrics.cue:512]: 👍

**Review by jszwedko — APPROVED** [review]
Thanks for circling back to this!

👍 pending upgrade guide.

File: website/cue/reference/components/sources/internal_metrics.cue (hunk 1 above)
Comment (001wwang) [reply website/cue/reference/components/sources/internal_metrics.cue:512]: Updated upgrade guide in dddf9c7. Let me know how it looks

**Review by jszwedko — APPROVED** [review]
Upgrade guide looks good!

**Pull request merged**